stdlib/public/core/HashedCollections.swift.gyb

```diff
@@ -2933,6 +2933,7 @@ struct _Native${Self}Storage<${TypeParametersDecl}> :
   }
 
   internal func assertingGet(_ i: Index) -> SequenceElement {
+    _precondition(i.offset >= 0 && i.offset < capacity)
```

Comment:
Is this fixing an existing mem-safety bug, or just compensating for the removal of asserts elsewhere?